docs: make |lspconfig-servers| jump to the markdown file

muniter opened the pull request:
By generating the tags at docgen and modifying the `|lspconfig-servers|` tag we can make `:help lspconfig-servers` take you to `./doc/server_configurations.md` and by adding a modeline to the markdown file we make sure it sets the correct filetype.

@mjlbach Let me know if you like this approach or you would like something different.

clason commented on doc/tags:
```diff
@@ -0,0 +1,26 @@
+buffer's	lspconfig.txt	/*buffer's*
```
typo: `buffers`?

muniter replied:
That is autogenerated by `:helptags` coming from this line https://github.com/neovim/nvim-lspconfig/blob/1c9c2120290d2fd6ea2cb55f954b5d108df0b628/doc/lspconfig.txt#L484 I'm guessing mj used that syntax to have some accent and it generates that on the tags.

clason replied:
Gotcha. (I'd have written that as `*buffer*'s`, but to each their own ;))

mjlbach replied:
Oh that tag shouldn't be there... Let's just change lspconfig.txt

mjlbach commented on doc/lspconfig.txt:
```diff
@@ -74,7 +74,7 @@ QUICKSTART                                                *lspconfig-quickstart*
 <
 - create a new project, ensure it contains a root marker which matches the
-  server requirements specified in `server_configurations.md`.
+  server requirements specified in |lspconfig-servers|.
```
lspconfig-server-configurations maybe?

mjlbach commented:
Great idea/improvement. Should we also mention this in the readme?

muniter commented:
Went ahead and added the tag in some places in the readme, and `server_configurations.md` itself for people that discover it first on github. Also changed a few things, all ears in corrections and things you prefere not to :)

muniter commented:
Also this is the logic that generates the tags in packer:
https://github.com/wbthomason/packer.nvim/blob/7f62848f3a92eac61ae61def5f59ddb5e2cc6823/lua/packer/plugin_utils.lua#L56-L73

Seems like as long as the tags modification time is later than the help file It won't regenerate the tags which should be the case here.

mjlbach commented on README.md:
````diff
@@ -82,7 +82,7 @@ require'lspconfig'.rust_analyzer.setup{}
 ```
 
-For a full list of servers, see [server_configurations.md](doc/server_configurations.md). This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed to setup. Most language servers can be i ...
+For a full list of servers, see [server_configurations.md](doc/server_configurations.md) `:help lspconfig-server-configurations`. This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed  ...
````
"accessible via"

mjlbach commented on README.md:
````diff
@@ -41,7 +41,7 @@ These features are not implemented in this repo, but in Neovim core. See `:help
     ```
 
-2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md)
+2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md) `:help lspconfig-server-configurations`
````
"accessible via"

mjlbach commented on doc/lspconfig.txt:
```diff
@@ -362,11 +365,12 @@ The three steps for adding and enabling a new server configuration are:
 <
 ==============================================================================
-INDEX                                                          *lspconfig-index*
+CONFIGURATIONS                                 *lspconfig-server-configurations*
```
won't the tag from this conflict with the tag from the tagfile? Should we just delete this section?

mjlbach commented on doc/lspconfig.txt:
```diff
@@ -479,8 +477,8 @@ that require a client id can either leverage tab-completion or the info
 - `:LspStart <config_name>` launches the requested (configured) client, but only
-  if it successfully resolves a root directory. Defaults to all *configured* 
-  servers matching the *current* *buffer's* filetype.
+  if it successfully resolves a root directory. Note Defaults to all
```
I would remove note here

mjlbach replied:
or add a comma/colon

mjlbach commented:
I think before you had (hit K to to got server-configurations or something of the like) which was useful

muniter commented:
The logic I'm thinking is: User needs to configure a servers does `:help lspconfig` looks for a configuration tag, first thing he sees is short paragraph telling him were they are. Sure can add the `K` thing.

mjlbach commented:
Yeah totally, just want it to be clear to less experienced users how to jump to a tag.

muniter commented:
Went with "type" since seems used a lot on `help`

mjlbach commented on doc/lspconfig.txt:
```diff
@@ -277,6 +279,9 @@ The global defaults for all servers can be overridden by extending the
 ==============================================================================
 SERVER CONFIGURATIONS                                 *lspconfig-configurations*
 
+See |lspconfig-server-configurations| by typing `K` over it for the complete
```
I would put this explanation in the first reference (intro) IMO

mjlbach commented on doc/lspconfig.txt:
```diff
@@ -74,10 +73,13 @@ QUICKSTART                                                *lspconfig-quickstart*
 
 - open a file within that project, such as `main.c`.
 
+- If you need more information about a server configuration read it's entry in
```
it's -> its

clason commented on README.md:
````diff
@@ -41,7 +41,7 @@ These features are not implemented in this repo, but in Neovim core. See `:help
     ```
 
-2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md)
+2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md) accessible via `:help lspconfig-server-configurations`
````
```suggestion
2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md). This list is also accessible via `:help lspconfig-server-configurations`.
```

clason commented on README.md:
````diff
@@ -82,7 +82,7 @@ require'lspconfig'.rust_analyzer.setup{}
 ```
 
-For a full list of servers, see [server_configurations.md](doc/server_configurations.md). This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed to setup. Most language servers can be i ...
+For a full list of servers, see [server_configurations.md](doc/server_configurations.md) accessible via `:help lspconfig-server-configurations`. This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in th ...
````
no need to repeat here? Or simply "see [...](...) or `:help ...`.

clason replied:
```suggestion
For a full list of servers, see [server_configurations.md](doc/server_configurations.md) or `:help lspconfig-server-configurations`. This document contains installation instructions and additional, optional, customization suggestions for each language server. For some servers that are not on your system path (e.g., `jdtls`, `elixirls`), you will be required to manually add `cmd` as an entry in the table passed to `setup`. Most language servers can be installed in less than a minute.
```

clason commented on doc/lspconfig.txt:
```diff
@@ -39,8 +38,8 @@ primary functionalities:
 
-See |lspconfig-index| for the reference to the complete list of language
-servers.
+See |lspconfig-server-configurations| for the reference to the complete list
+of language servers.
```
```suggestion
See |lspconfig-server-configurations| for the complete list of language servers.
```

clason commented on doc/lspconfig.txt:
```diff
@@ -74,10 +73,13 @@ QUICKSTART                                                *lspconfig-quickstart*
     require'lspconfig'.clangd.setup{}
 <
 - create a new project, ensure it contains a root marker which matches the
```
```suggestion
- create a new project, ensure that it contains a root marker which matches the
```

clason commented on doc/lspconfig.txt:
```diff
@@ -218,7 +220,7 @@ passed overrides to `setup {}` are:
   The script that automatically generates `server_configurations.md` converts
   the `package.json` referenced in a server configuration source file
-  into a list of optional settings listed in `server_configurations.md`.
+  into a list of optional settings listed in |lspconfig-server-configurations|
```
```suggestion
  into a list of optional settings listed in |lspconfig-server-configurations|.
```

clason commented on doc/lspconfig.txt:
```diff
@@ -74,10 +73,13 @@ QUICKSTART                                                *lspconfig-quickstart*
 
 - open a file within that project, such as `main.c`.
 
+- If you need more information about a server configuration read it's entry in
+  |lspconfig-server-configurations|
```
```suggestion
- If you need more information about a server configuration, read the corresponding
  entry in |lspconfig-server-configurations|.
```

mjlbach commented:
Couldn't push to the branch so committed clason's changes in https://github.com/neovim/nvim-lspconfig/pull/1497
